src/acc/libsmm_acc/libcusmm/libcusmm.cpp

```diff
@@ -207,41 +211,15 @@ int libcusmm_process_d(int *param_stack, int stack_size, CUstream stream, int m,
 
```

Comment:
This read access to `kernel_handles` is still unprotected.

Reply:
True. But does it really have to be? 
The result of this lookup could be:
- `h_mnk` is found in `kernel_handles`: in which case we can just get the kernel handle and use it
- `h_mnk` is *not* found in `kernel_handles`: in which case the thread will (wait and) enter the critical section in which accesses to `kernel_handles` are protected. By the time it enters the critical section, the kernel might have been jitted already, or not. 

I don't see what problems could arise?

Reply:
The problem arises when another thread calls `find` while the first threads calls `emplace`. During the emplace call the map might be in an inconsistent state, e.g. in the middle of rebalancing itself.